### PR TITLE
[12.0][ADD] web_set_single_page_hidden

### PR DESCRIPTION
**Pull request by vvrossem**
In 12.0, on a _PO_, _SO_, _WH/IN_, _WH/OUT_, or, more generally, anything that has _lines_ in its `form` view uses the `Pager` widget to visualise the line total count  (i.e. `1-10/10`).

`Pager` has the following `@param {boolean} [options.single_page_hidden] (not) to display the pager if only one page`.
On `init` function, the `single_page_hidden` option is set to `false`.
On `_render` function, the `Pager` is hidden `if (size === 0 || (this.options.single_page_hidden && this._singlePage()))`

`FieldX2Many`, an extension of `AbstractField`, instantiates a `Pager` with `option` `single_page_hidden: true,` in its `_renderControlPanel` function. It's impossible to `require` this field since it has not been made available (i.e no `return { FieldX2Many: FieldX2Many}`) in Odoo.

Two relational fields - `FieldOne2Many` & `FieldMany2Many` - extends `FieldX2Many`.
Their `start` function is overriden to set `single_page_hidden = false` on their `Pager` widget.

Before: (without `1-4/4`)
before

After: (with `1-4/4`)
after

### CODE REVIEW

**Review by houssine78 — APPROVED** [review]

File: web_set_single_page_hidden/static/src/js/relational_fields.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    });
+
+    FieldMany2Many.include({
+        start: function () {
+            var self = this;
+            return this._super.apply(this, arguments).then(function () {
+                if (!_.isUndefined(self.pager)) {
+                    self.pager.set_single_page_hidden(false);
+                }
+            });
```
Comment (hparfr) [inline web_set_single_page_hidden/static/src/js/relational_fields.js:26]: I don't get why you bind this to _super and don't bind it in the promise.

File: web_set_single_page_hidden/static/src/js/pager.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+odoo.define('web_set_single_page_hidden.Pager', function (require) {
+    "use strict";
+
+    var Pager = require('web.Pager');
+
+    Pager.include({
+        set_single_page_hidden: function (value) {
+            if (!_.isUndefined(this.options)) {
```
Comment (hparfr) [inline web_set_single_page_hidden/static/src/js/pager.js:8]: why use _.isUndefined ? 

```suggestion
            if (this.options) {
```

**Review by hparfr — APPROVED** [review]
I don't get the use case.

I have few minor non-blocking remarks in the JS.

**Comment by vvrossem** [comment]
@hparfr thank you for reviewing this PR :)

The use case is to force the display of the line total count. In standard Odoo, the Pager is hidden when - I think - there are < 40 items.

File: web_set_single_page_hidden/static/src/js/pager.js (hunk 2 above)
Comment (vvrossem) [reply web_set_single_page_hidden/static/src/js/pager.js:8]: Indeed, your suggestion has better readability. I'll change it :)

File: web_set_single_page_hidden/static/src/js/relational_fields.js (hunk 1 above)
Comment (vvrossem) [reply web_set_single_page_hidden/static/src/js/relational_fields.js:26]: I'm not sure to understand what you mean. How to properly call parent function, which is:
```js
    start: function () {
        return this._renderControlPanel().then(this._super.bind(this));
    },
```

**Comment by vvrossem** [comment]
@houssine78  thanks for the review!

**Comment by houssine78** [comment]
Hi @OCA/web-maintainers some of you could have a look a this PR ? It brings back a small feature from 9.0

**Comment by hparfr** [comment]
> The use case is to force the display of the line total count. In standard Odoo, the Pager is hidden when - I think - there are < 40 items.

It's not a use case, it's a description. My question can be rephrased :  " **why** you need to always display the pager ?"

**Comment by vvrossem** [comment]
We need to always display the pager so the user doesn't have to count the items present in the document one by one

**Review by legalsylvain — APPROVED** [review]
Code review. no test. thanks !

**Comment by vvrossem** [comment]
@legalsylvain Thank you for the review :)

**Comment by MeetKD** [comment]
@vvrossem I would suggest to add a system parameter like display single page pager or something and check the condition in js accordingly.

**Comment by houssine78** [comment]
Hi @pedrobaeza I think this PR is ready for merge.

**Comment by houssine78** [comment]
many thanks @pedrobaeza  !

**Comment by vvrossem** [comment]
Thank you everybody for your time reviewing this PR. Thank you @pedrobaeza for merging it !

**Pull request merged**